src/libs/antares/study/parameters.h

```diff
@@ -258,6 +258,8 @@ class Parameters final
     uint nbTimeSeriesWind;
     //! Nb of timeSeries : Thermal
     uint nbTimeSeriesThermal;
+    //! Nb of timeSeries : Renewable
+    uint nbTimeSeriesRenewable;
```

Comment:
>  And a question is : if thermal has ready made TS, how does the TS number are drawn ?

Here is the code when expanding the `DRAW_A_RANDOM_NUMBER` macro.

```cpp
if (intramodal[TS_INDEX(T)]) {
  cluster->series->timeseriesNumbers.entry[0][y] = draw[TS_INDEX(T)];
 } else {
  if (tsgen[TS_INDEX(T)]) {
    cluster->series->timeseriesNumbers.entry[0][y] = (uint32)(floor((double)(study.runtime->random[Data::seedTimeseriesNumbers].next()) * parameters.nbTimeSeriesThermal));
  } else {
    cluster->series->timeseriesNumbers.entry[0][y] = (uint32)(floor((double)(study.runtime->random[Data::seedTimeseriesNumbers].next()) * cluster->series->series.width));
  }
 }
```
When `tsgen[TS_INDEX(T)]=false`, TS generation is disabled, the ready-made series width is used.

I agree that since no TS generation is possible for renewable clusters, we should **always** use `cluster->series->series.width`.

Reply:
Note that
```cpp
if (intramodal[z])
{
    draw[z] = (uint32)(floor(study.runtime->random[Data::seedTimeseriesNumbers].next()
                             * nbTimeseries[z]));
}
```
where `nbTimeseries[z]` is obtained with function `CheckMatricesWidth` : 
- N if all matrices are of width N >= 1
- 1 if all matrices are of width 0
- 0 in error cases (inconsistent width)